src/intersection/compute_intersection.cc

```diff
@@ -281,54 +285,86 @@ bool ComputeIntersection< Simplex< 1  >, Simplex< 2  > >::compute_pathologic(uns
+    unsigned int zero_idx_sum =0;
+
+    for (unsigned int i=0; i < 3; i++) {
+        //DebugOut().VarFmt(i).VarFmt(w[i]);
+        if (w[i] > rounding_epsilon) n_positive++;
+        else if ( w[i] > -rounding_epsilon) zero_idx_sum+=i;
+        else n_negative++;
+    }
+
+    // any negative barycentric coordinate means, no intersection
+    if (n_negative>0) return IntersectionResult::none;
+
+    // test whether any plucker products is non-zero
+    if (n_positive > 0) {
         IntersectionPointAux<1,2> IP;
         
-        if(compute_plucker(IP, w))
-        {
-            IP12s.push_back(IP);
-            return true;
+        compute_plucker(IP, w);
+        // edge of triangle
+        unsigned int non_zero_idx=0;
+        if (n_positive == 2) {
+            // one zero product, intersection on the zero edge
+            // the zero edge index is equal to zero_idx_sum
+            IP.set_topology_B(zero_idx_sum, 1);
+            non_zero_idx =  (zero_idx_sum + 1) % 3;
         }
-    }else if(compute_zeros_plucker_products){
-        
-        // 1 zero product -> IP is on the triangle side
-        // 2 zero products -> IP is at the vertex of triangle (there is no other IP)
+        else if (n_positive == 1) {
+            // two zero products, intersection in vertex oposite to single non-zero edge
+            // index of the non-zero edge is 3-zero_idx_sum
+            IP.set_topology_B(RefElement<2>::oposite_node(3-zero_idx_sum), 0);
+            non_zero_idx = 3-zero_idx_sum;
+        }
+        //DebugOut().VarFmt(non_zero_idx).VarFmt(signed_plucker_product(non_zero_idx));
+
+        auto result = signed_plucker_product(non_zero_idx) > 0 ?
+                IntersectionResult::positive : IntersectionResult::negative;
+        IP.set_orientation(result);
+
+        IP12s.push_back(IP);
+        return result;
+
+    } else if(compute_zeros_plucker_products){
+        ASSERT_DBG(0).error("currently unsupported");
         // 3 zero products: 
         //      -> IP is at the vertex of triangle but the line is parallel to opossite triangle side
         //      -> triangle side is part of the line (and otherwise)
         for(unsigned int i = 0; i < 3;i++){
-            if( std::abs(w[i]) <= rounding_epsilon ){
-//                 DBGMSG("Intersections - Pathologic case, side %d.\n",i);
                 IntersectionPointAux<1,2> IP;
                 if(compute_pathologic(i,IP))
                 {
-//                     DBGMSG("Intersections - Pathologic case PUSH.\n");
                     IP12s.push_back(IP);
-                    return true;
+                    return IntersectionResult::degenerate;
                 }
-            }
         }
+    } else {
+        return IntersectionResult::degenerate;
     }
-    
-    return false;   // if IP not found before
 };
 
 unsigned int ComputeIntersection< Simplex< 1  >, Simplex< 2  > >::compute_final(vector< IntersectionPointAux< 1, 2 > >& IP12s)
```

Comment:
function compute_final differs from compute
I think it should use the function compute, then possibly cut the abscissa; "compute_pathologic" then has to deal only with the degenerate case;